### PR TITLE
validate nil attachment POST param

### PR DESCRIPTION
**Pull request by omgitsbillryan**
## Description of change
Adds basic param validation to all controllers using the `FormAttachmentCreate` concern. 

The motivation for this PR was a [`NoMethodError` in Sentry](http://sentry.vfs.va.gov/vets-gov/platform-api-production/issues/90775/?referrer=slack) where nested `params` parameters were being read w/o the safe-navigation operator or a validation that the param existed.

Connects: https://github.com/department-of-veterans-affairs/vets-api/pull/3721

## Testing done
Specs

## Testing planned
None.

#### Unique to this PR

- [ ] _Replace this line with individual tasks unique to your PR_

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: app/controllers/concerns/form_attachment_create.rb [hunk 1]
```diff
@@ -11,7 +11,10 @@ def create
     form_attachment_model = self.class::FORM_ATTACHMENT_MODEL
     form_attachment = form_attachment_model.new
     namespace = form_attachment_model.to_s.underscore.split('/').last
-    form_attachment.set_file_data!(params[namespace][:file_data])
+
+    form_attachment_params = params.require(namespace).permit(:file_data)
```
Comment (johnpaulashenfelter) [inline app/controllers/concerns/form_attachment_create.rb]: Might something like `attached_form` read better? There's only one param so params doesn't read well to me. 

```
attached_form = params.require(namespace).permit(:file_data)
form_attachment.set_file_data!(attached_form)
```

Maybe even `file_data` to "hoist" that param name up to the top level

```
file_data = params.require(namespace).permit(:file_data)
form_attachment.set_file_data!(file_data)
```

Comment (omgitsbillryan) [reply app/controllers/concerns/form_attachment_create.rb]: 👍

**Comment by omgitsbillryan** [comment]
@johnpaulashenfelter you're right, so I went ahead and added the swagger `400` response + spec for  `hca_attachments `.

I'm afraid `upload_supporting_evidences` & `supporting_documentation_attachements` both lack any swagger at all 😬 , which explains why no specs needed updating. @lihanli is it worth logging a ticket to add this swagger? I'm not even sure if swagger existed when this controller was created.

**Comment by lihanli** [comment]
@omgitsbillryan creating an issue for the swagger shouldn't hurt

**Comment by omgitsbillryan** [comment]
Issue created : https://github.com/department-of-veterans-affairs/va.gov-team/issues/4634

**Pull request merged**